DQMOffline/Trigger/plugins/LepHTMonitor.cc

```diff
@@ -0,0 +1,446 @@
+  h_lepEtaTurnOn_den_(nullptr),
+  h_lepPhiTurnOn_num_(nullptr),
+  h_lepPhiTurnOn_den_(nullptr),
+  h_NPVTurnOn_num_(nullptr),
+  h_NPVTurnOn_den_(nullptr)
+  {
+  edm::LogInfo("LepHTMonitor")
+    << "Constructor LepHTMonitor::LepHTMonitor\n";
+  }
+
+LepHTMonitor::~LepHTMonitor(){
+  edm::LogInfo("LepHTMonitor")
+    << "Destructor LepHTMonitor::~LepHTMonitor\n";
+  if (num_genTriggerEventFlag_) delete num_genTriggerEventFlag_;
+  if (den_lep_genTriggerEventFlag_) delete den_lep_genTriggerEventFlag_;
+  if (den_HT_genTriggerEventFlag_) delete den_HT_genTriggerEventFlag_;
+}
+
+void LepHTMonitor::dqmBeginRun(const edm::Run &run, const edm::EventSetup &e){
+  edm::LogInfo("LepHTMonitor") << "LepHTMonitor::beginRun\n";
+}
+
+void LepHTMonitor::bookHistograms(DQMStore::IBooker &ibooker,
+                                           const edm::Run &iRun, const edm::EventSetup &iSetup){
+  edm::LogInfo("LepHTMonitor") << "LepHTMonitor::bookHistograms\n";
+  //book at beginRun
+  ibooker.cd();
+  ibooker.setCurrentFolder("HLT/LepHT/" + folderName_);
+ 
+
+  bool is_mu = false;
+  bool is_ele = false;
+  if(theElectronTag_.label() == "" && theMuonTag_.label() != ""){
+    is_mu=true;
+  }else if(theElectronTag_.label() != "" && theMuonTag_.label() == ""){
+    is_ele=true;
+  }
+  //Cosmetic axis names
+  std::string lepton="lepton", Lepton="Lepton";
+  if(is_mu && !is_ele){
+    lepton="muon";
+    Lepton="Muon";
+  }else if(is_ele && !is_mu){
+    lepton="electron";
+    Lepton="Electron";
+  }
+  //Initialize trigger flags
+  if ( num_genTriggerEventFlag_ && num_genTriggerEventFlag_->on() ) num_genTriggerEventFlag_->initRun( iRun, iSetup );
+  if ( den_lep_genTriggerEventFlag_ && den_lep_genTriggerEventFlag_->on() ) den_lep_genTriggerEventFlag_->initRun( iRun, iSetup );
+  if ( den_HT_genTriggerEventFlag_ && den_HT_genTriggerEventFlag_->on() ) den_HT_genTriggerEventFlag_->initRun( iRun, iSetup );
+  
+
+  //Convert to vfloat for picky TH1F constructor
+  vector<float> f_ptbins;
+  for(uint ib=0;ib<ptbins_.size();ib++) f_ptbins.push_back(static_cast<float>(ptbins_[ib]));
+  vector<float> f_htbins;
+  for(uint ib=0;ib<htbins_.size();ib++) f_htbins.push_back(static_cast<float>(htbins_[ib]));
+
+  //num and den hists to be divided in harvesting step to make turn on curves
+  h_pfHTTurnOn_num_ = ibooker.book1D("pfHTTurnOn_num", "Numerator;Offline H_{T} [GeV];",  f_htbins.size()-1, f_htbins.data());
+  h_pfHTTurnOn_den_ = ibooker.book1D("pfHTTurnOn_den","Denominator;Offline H_{T} [GeV];",  f_htbins.size()-1, f_htbins.data());
+  
+  h_lepPtTurnOn_num_  = ibooker.book1D("lepPtTurnOn_num", ("Numerator;Offline "+lepton+" p_{T} [GeV];").c_str(),   f_ptbins.size()-1, f_ptbins.data());
+  h_lepPtTurnOn_den_  = ibooker.book1D("lepPtTurnOn_den", ("Denominator;Offline "+lepton+" p_{T} [GeV];").c_str(), f_ptbins.size()-1, f_ptbins.data());
+  h_lepEtaTurnOn_num_ = ibooker.book1D("lepEtaTurnOn_num", "Numerator;Offline lepton #eta;", 10,-2.5,2.5);
```

Comment:
I think we already discuss about this
please, do not make use of hard coded values